packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -171,7 +181,13 @@ class ChangeNotifier implements Listenable {
   @override
   void removeListener(VoidCallback listener) {
     assert(_debugAssertNotDisposed());
-    _listeners!.remove(listener);
+    _listeners
+        !.cast<_ListenerEntry?>()
+        .firstWhere(
+          (_ListenerEntry? element) => element!.listener == listener,
+          orElse: () => null,
+        )
+        ?.unlink();
```

Comment:
Wondering if this would be even faster with a regular for loop. (The code would definitely be prettier without the casting).